### ISSUE TITLE
Cannot Paste Image From Clipboard

### ISSUE THREAD

**Issue by BarDev** [lostintangent/gistpad#135]
While using VSCode, when I attempt to paste an image into a markdown (.md) file, I receive the following error messages. I've copied images from multiple sources, including images from Chrome and Snag-It.

![image](https://user-images.githubusercontent.com/1308620/76466506-77e1cc00-63ad-11ea-8ad3-2b97a7a8bc2c.png)

> There doesn't appear to be an image on your clipboard...

![image](https://user-images.githubusercontent.com/1308620/76466103-94c9cf80-63ac-11ea-9c8e-61185dd77316.png)

GistPad Version: .0.0.62
VSCode Version: 
![image](https://user-images.githubusercontent.com/1308620/76466268-ec683b00-63ac-11ea-8877-a3afa5467d8d.png)

I know images are being copied to the clipboard because the images above were pasted from the clipboard.

**Comment by lostintangent** [comment]
@BarDev Thanks for reporting this! Could you confirm how you’re copying images from Chrome? Right-clicking them and selecting “Copy Image”?

**Comment by BarDev** [comment]
@lostintangent Yes, That's correct.  I right-click the image and from the menu I select "Copy Image".

![image](https://user-images.githubusercontent.com/1308620/76703758-8ecc3b00-6699-11ea-91ad-74c2d5dea3e2.png)

**Comment by lostintangent** [comment]
Hmm, interesting. I'm not able to repro this, and I'm also using Chrome on macOS. @legomushroom Any thoughts here?

**Comment by legomushroom** [comment]
I don't think the scenario above is supported by the feature, it generally works only with the images that are taken by the built-in screen capture screenshots. https://support.apple.com/en-us/HT201361

**Comment by BarDev** [comment]
Based on @legomushroom comment, I was successful with pasting an image into VSCode Gist by doing a screen capture using the MacOs built-in screen capture and directing it to the clipboard.

For completeness, I was unsuccessful in pasting an image from copying the image from the browser and coping image from 3rd party app.

**Comment by lostintangent** [comment]
Closing this as resolved per the previous comment 👍

**Comment by shuivin** [comment]
same issue in win10.
built-in screen capture, chrome copy image , no work.

**Comment by IlanVivanco** [comment]
Getting the same problem on Win10. It does not work either by using the screen capture tool nor by copy images from chrome.
![image](https://user-images.githubusercontent.com/9307840/108203712-a3168480-7101-11eb-982b-695dfb68873f.png)

I tested it, and if using the `Gistpad > Images: Paste Type` setting to `base64` does in fact paste the image correctly. But when going back to `file` breaks again.

Also tested this on both side, Windows and WSL.

**Comment by IlanVivanco** [comment]
I just realize the images are being uploaded to Gist, despite the fact that the error is shown.
However, the image URL is not being placed correctly on the `.md`

![image](https://user-images.githubusercontent.com/9307840/108206469-2f767680-7105-11eb-809b-1d54c0b4e28b.png)

**Comment by carlocardella** [comment]
Sorry to resurrect a closed thread: I had the same problem on Windows 10 but after some experimenting I realized this happens when there are multiple files under the same Gist: 
![image](https://user-images.githubusercontent.com/5784415/110397466-bfce1a00-8026-11eb-9c69-26dc0ae372b3.png)
The upload works fine for single-file Gists and I see the `Images` folder being created:
![image](https://user-images.githubusercontent.com/5784415/110397544-e724e700-8026-11eb-8f2c-3567fd8b4ce5.png)

**Comment by Fed03** [comment]
Same problem described by others. Using Win10

**Comment by lostintangent** [comment]
I'm re-opening this issue since the issue seems to be

**Comment by Berkays** [comment]
I managed a workaround for windows. See #214

**Comment by carlocardella** [comment]
Works for me now, thanks

**Comment by lostintangent** [comment]
Awesome, thanks for confirming that fix! And thanks @Berkays for the contribution 😄

**Comment by IlanVivanco** [comment]
@Berkays It does work for me too on my gists, but I'm getting a 404 when using scratch notes. Different problem though, the original problem seems to be solved.

![image](https://user-images.githubusercontent.com/9307840/112975692-570c2600-912a-11eb-8fae-25292c6bbeeb.png)

<details>
<summary>Console logs</summary>
<pre>
ERR undefined is not iterable (cannot read property Symbol(Symbol.iterator)): TypeError: undefined is not iterable (cannot read property Symbol(Symbol.iterator))
    at c.provideDocumentLinks (c:\Users\melio\.vscode\extensions\vsls-contrib.gistfs-0.2.7\out\prod\extension.js:319:187727)
    at c:\Users\melio\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:87:171064
    at c:\Users\melio\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:57:7538
    at new Promise (<anonymous>)
    at Object.g [as asPromise] (c:\Users\melio\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:57:7510)
    at $.provideLinks (c:\Users\melio\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:87:171035)
    at c:\Users\melio\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:87:187672
    at Se._withAdapter (c:\Users\melio\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:87:176060)
    at Se.$provideDocumentLinks (c:\Users\melio\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:87:187648)
    at c._doInvokeHandler (c:\Users\melio\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:90:10509)
    at c._invokeHandler (c:\Users\melio\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:90:10201)
    at c._receiveRequest (c:\Users\melio\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:90:8820)
    at c._receiveOneMessage (c:\Users\melio\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:90:7673)
    at c:\Users\melio\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:90:5782
    at g.fire (c:\Users\melio\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:57:1836)
    at p.fire (c:\Users\melio\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:65:15443)
    at c:\Users\melio\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:106:29119
    at g.fire (c:\Users\melio\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:57:1836)
    at p.fire (c:\Users\melio\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:65:15443)
    at t._receiveMessage (c:\Users\melio\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:65:20693)
    at c:\Users\melio\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:65:17587
    at g.fire (c:\Users\melio\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:57:1836)
    at l.acceptChunk (c:\Users\melio\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:65:12808)
    at c:\Users\melio\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:65:12156
    at Socket.E (c:\Users\melio\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:106:12375)
    at Socket.emit (events.js:315:20)
    at addChunk (_stream_readable.js:295:12)
    at readableAddChunk (_stream_readable.js:271:9)
    at Socket.Readable.push (_stream_readable.js:212:10)
    at Pipe.onStreamRead (internal/stream_base_commons.js:186:23)
</pre>
</details>

**Comment by Berkays** [comment]
I couldn't reproduce this issue. Are there any specific steps that i can follow? @IlanVivanco

**Comment by IlanVivanco** [comment]
Here is a screencast for you to see what I'm doing.

https://user-images.githubusercontent.com/9307840/114074526-d7ddc700-987a-11eb-8439-fa0243f2977a.mp4

I'm still getting an error on the console constantly (not just when pasting images).

```log
ERR undefined is not iterable (cannot read property Symbol(Symbol.iterator)): TypeError: undefined is not iterable (cannot read property Symbol(Symbol.iterator))
    at c.provideDocumentLinks (c:\Users\melio\.vscode\extensions\vsls-contrib.gistfs-0.2.8\out\prod\extension.js:319:187767)
    at c:\Users\melio\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:86:174320
    at c:\Users\melio\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:64:12046
    at new Promise (<anonymous>)
    at g (c:\Users\melio\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:64:12018)
    at V.provideLinks (c:\Users\melio\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:86:174300)
    at c:\Users\melio\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:86:190957
    at Ie._withAdapter (c:\Users\melio\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:86:179343)
    at Ie.$provideDocumentLinks (c:\Users\melio\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:86:190933)
    at u._doInvokeHandler (c:\Users\melio\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:90:12794)
    at u._invokeHandler (c:\Users\melio\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:90:12478)
    at u._receiveRequest (c:\Users\melio\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:90:11094)
    at u._receiveOneMessage (c:\Users\melio\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:90:9931)
    at c:\Users\melio\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:90:8032
    at _.fire (c:\Users\melio\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:57:1836)
    at E.fire (c:\Users\melio\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:65:15496)
    at c:\Users\melio\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:105:30083
    at _.fire (c:\Users\melio\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:57:1836)
    at E.fire (c:\Users\melio\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:65:15496)
    at t._receiveMessage (c:\Users\melio\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:65:20754)
    at c:\Users\melio\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:65:17640
    at _.fire (c:\Users\melio\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:57:1836)
    at acceptChunk (c:\Users\melio\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:65:12861)
    at c:\Users\melio\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:65:12209
    at Socket.w (c:\Users\melio\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:105:13173)
    at Socket.emit (events.js:315:20)
    at Socket.EventEmitter.emit (domain.js:483:12)
    at addChunk (_stream_readable.js:295:12)
    at readableAddChunk (_stream_readable.js:271:9)
    at Socket.Readable.push (_stream_readable.js:212:10)
    at Pipe.onStreamRead (internal/stream_base_commons.js:186:23)
```

I'm not having the same problems as in https://github.com/lostintangent/gistpad/issues/212#issue-820300667, but the error on the console seems pretty similar to me. Maybe we should open another issue for that particular warning.

**Comment by Fed03** [comment]
same problem as @IlanVivanco but with all type of notes, normal or scratch

**Comment by roboticsmick** [comment]
Same problem for me using Ubuntu 20.04 + VSCode + GISTPAD.

The default screenshot settings all fail for me:

- shift + ctrl + printsceen
- ctrl + printscreen
- right click => copy image

The images paste fine into other applications though.

![image](https://user-images.githubusercontent.com/70121687/127733937-9a537606-3308-4403-9175-bbc95398949d.png)

I get this whenever I right click, and select paste image in GISTPAD.

**Comment by roboticsmick** [comment]
> Same problem for me using Ubuntu 20.04 + VSCode + GistPad.
> 
> The default screenshot settings all fail for me:
> 
>     * shift + ctrl + printsceen
> 
>     * ctrl + printscreen
> 
>     * right click => copy image
> 
> 
> The images paste fine into other applications though.
> 
> ![image](https://user-images.githubusercontent.com/70121687/127733937-9a537606-3308-4403-9175-bbc95398949d.png)
> 
> I get this whenever I right click, and select paste image in GistPad.

Scrap that. It seems to work. I don't know why, but the only things I have done since my last post was disabling two extensions I also had running for GistHub (Markdown All in One and markdownlint). Worked when I disabled them. I re-enabled them though, and it's still working, and I also logged in and out? 

Love the app and it works with screenshots so I'm stoked!

This is the perfect application for taking coding notes in uni lectures! Thanks so much!

**Comment by lostintangent** [comment]
Thanks for confirming that you were able to get unblocked! Let me know if you run into any other issues or have feedback on how things can be further improved 🚀

**Comment by AzimsTech** [comment]
I ran into the same issue as @IlanVivanco.

![gif](https://i.imgur.com/UHZ2sFJ.gif)

**Comment by IlanVivanco** [comment]
> I ran into the same issue as @IlanVivanco.
> 
> ![gif](https://camo.githubusercontent.com/591b53da67991633f01297b7db2f8b9120ce9edeafaf58c901a1ba35992e01f7/68747470733a2f2f692e696d6775722e636f6d2f55485a3273464a2e676966)

It's been working wonderfully for me since then.

**Comment by AzimsTech** [comment]
@IlanVivanco Any idea what's causing this?

**Comment by IlanVivanco** [comment]
@AzimsTech if you go to the top menu to "Help > Toggle Developer Tools" and then to the "Console" tab, you may find some errors that will help to find a possible reason.

**Comment by AzimsTech** [comment]
@IlanVivanco I've checked it. It's the same error as [yours.](https://github.com/lostintangent/gistpad/issues/135#issuecomment-816026877)

**Comment by lostintangent** [comment]
Just to make sure I can try to repro the issue: which OS are you on? And are you using desktop VS Code or the web client (e.g. GitHub.dev)?

**Comment by AzimsTech** [comment]
@lostintangent  Windows 10 Enterprise 21H2 & VS Code desktop & VSCodium. Also on Ubuntu 21.10

**Comment by ng-hai** [comment]
Seems like it does not work on my Mac M1 too

https://user-images.githubusercontent.com/19520278/172151583-b4dff87c-fd44-4b89-a61c-5e49c7fc5a4c.mov

**Comment by Warns** [comment]
I'm getting a similar error with image being broken (404) after upload. Tried the methods mentioned but all upload it as broken.

![image](https://user-images.githubusercontent.com/4021052/183244215-5540ce6a-2152-4ff8-856d-8dbb986a6d65.png)

**Comment by YanZheng-16** [comment]
same issue in win11.
I disable all the markdown extensions of vs code, use the default screenshot, and use the Gistpad > Images: Paste Type setting to base64. But none of the above works. The image is still broken (404) after upload.

**Comment by YanZheng-16** [comment]
It is interesting that if I use the Gistpad > Images: Paste Type setting to base64, I can see the images in vs code but in Chrome, it will show Not Found (no 404).
